File: Source/Fayde.Server/Media/Animation/ObjectAnimationUsingKeyFrames.cs
using Fayde.Core;
using Fayde.Xaml.Metadata;

namespace Fayde.Media.Animation
{
    public class ObjectAnimationUsingKeyFrames : Timeline
    {
        public static readonly PropertyDescription KeyFrames = PropertyDescription.Register("KeyFrames", typeof(DependencyObjectCollection<ObjectKeyFrame>), typeof(ObjectAnimationUsingKeyFrames), true);
    }
}

File: src/Media/Animation/ObjectAnimationUsingKeyFrames.ts
/// <reference path="AnimationUsingKeyFrames.ts" />

module Fayde.Media.Animation {
    export class ObjectAnimationUsingKeyFrames extends AnimationUsingKeyFrames {
        Resolve(target: DependencyObject, propd: DependencyProperty): boolean {
            var enumerator = this.KeyFrames.getEnumerator();
            while (enumerator.moveNext()) {
                var keyFrame = <ObjectKeyFrame>enumerator.current;
                var value = keyFrame.Value;
                if (value == null) {
                    keyFrame.ConvertedValue = undefined;
                } else {
                    var converted: any;
                    try {
                        converted = nullstone.convertAnyToType(value, <Function>propd.GetTargetType());
                    } catch (err) {
                        console.warn("Error resolving ObjectAnimation Value.");
                        return false;
                    }
                    keyFrame.ConvertedValue = converted;
                }
            }
            return super.Resolve(target, propd);
        }
    }
    Fayde.CoreLibrary.add(ObjectAnimationUsingKeyFrames);
}